gems/aws-sdk-core/lib/aws-sdk-core/credentials_provider_chain.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  class CredentialsProviderChain
+    def call(config)
+      providers.each do |method_name|
+        provider = send(method_name, config)
+        return provider if provider
+      end
+      nil
+    end
+
+    private
+
+    def providers
+      [
+        :stubbed_credentials,
+        # in-code credentials are resolved by Config
+        :env_static_credentials,
+        :env_assume_role_web_identity_credentials,
```

Comment:
is there not env_assume_role_credentials?

Reply:
Not that I'm aware. I think AWS_ROLE_ARN should apply here but we have no such customization.